graphql_compiler/query_formatting/cypher_formatting.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import datetime
+import json
+from string import Template
+
+import arrow
+from graphql import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLInt, GraphQLList, GraphQLString
+import six
+
+from ..compiler import CYPHER_LANGUAGE
+from ..compiler.helpers import strip_non_null_from_type
+from ..exceptions import GraphQLInvalidArgumentError
+from ..schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
+from .representations import coerce_to_decimal, represent_float_as_str, type_check_and_str
+
+
+def _safe_cypher_string(argument_value):
+    """Sanitize and represent a string argument in Cypher."""
+    if not isinstance(argument_value, six.string_types):
+        if isinstance(argument_value, bytes):  # should only happen in py3
+            argument_value = argument_value.decode('utf-8')
+        else:
+            raise GraphQLInvalidArgumentError(u'Attempting to convert a non-string into a string: '
+                                              u'{}'.format(argument_value))
+
+    # Using JSON encoding means that all unicode literals and special chars
+    # (e.g. newlines and backslashes) are replaced by appropriate escape sequences.
+    # However, the quoted result is wrapped in double quotes, and $ signs are not escaped,
+    # so that would allow arbitrary code execution in Cypher.
+    # TODO: Leon
+    #  does the gremlin strategy work? alternatively, just wrap everything in backticks. But then
```

Comment:
This seems to be quite important for security purposes so I won't merge the branch until this is addressed. Could you please figure this out? Maybe ask Predrag when he is unoccupied. Though it could be faster to figure this out by yourself. The rest of the branch looks good. Also please do not pose question TODOs like this one in the future. It's really a question. Not a TODO.

Reply:
Went over it with Predrag. Cypher doesn't allow arbitrary code execution in its strings that way Groovy [does](http://www.groovy-lang.org/syntax.html#_double_quoted_string) so we're good to go. Dates and datetime formatting is fine as well.